### DISCUSSION TITLE
How to do integration test with MySQLContainer and MyBatis

### DISCUSSION THREAD

**Discussion by xenogew** [testcontainers/testcontainers-java#5025] [Q&A] [answered]
I want to start with my implement scenario.
Spring Boot
Data Source: dual datasource, load jdbc config from `application.yml` properties file

I've read other discussion which suggest to use TestContainer with `tc` JDBC URL scheme, but I've tried with 3 examples from Google result, but it disappointed with exception on loading.

These are exception messages
```
org.mybatis.spring.MyBatisSystemException: nested exception is org.apache.ibatis.exceptions.PersistenceException: 
### Error querying database.  Cause: java.lang.IllegalArgumentException: jdbcUrl is required with driverClassName.
### The error may exist in file [C:\workspace\package\to\MyDataDao.xml]
### The error may involve package.to.MyDataDao.findListOf
### The error occurred while executing a query
### Cause: java.lang.IllegalArgumentException: jdbcUrl is required with driverClassName.
```

Or any others suggestion would be appreciate.

This is my implementation

```
public class TestContainersContextInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext>
  MySQLContainer<?> MYSQL_CONTAINER = new MySQLContainer<>(DockerImageName.parse(MYSQL_IMAGE));
  @Override
    public void initialize(@NotNull ConfigurableApplicationContext applicationContext) {
        var jdbcUrl = MYSQL_CONTAINER.getJdbcUrl();
        TestPropertyValues.of("my.custom.datasource.url=", jdbcUrl)
                .applyTo(applicationContext.getEnvironment());
    }

// then on Test class annotate with
@ContextConfiguration(initializers = {TestContainersContextInitializer.class})
```

**Comment by xenogew** [comment, marked as answer]
I found solution now.

I used here
https://github.com/wearearima/spring-data-jdbc-testcontainers/blob/master/src/test/java/eu/arima/springdatajdbctestcontainers/AccountRepositoryTest.java#L94

Sorry guys for trouble your space.

**Reply by kiview** [reply]
Note that this approach is using Testcontainers with instantiated objects, rather than using the special JDBC-URL support as documented here: https://www.testcontainers.org/modules/databases/jdbc/
